drivers/bluetooth/hci/rpmsg.c

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <init.h>
+#include <sys/byteorder.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/hci_driver.h>
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_HCI_DRIVER)
+#define LOG_MODULE_NAME bt_hci_driver
+#include "common/log.h"
+
+#define RPMSG_CMD 0x01
+#define RPMSG_ACL 0x02
+#define RPMSG_SCO 0x03
+#define RPMSG_EVT 0x04
+
+int bt_rpmsg_platform_init(void);
+int bt_rpmsg_platform_send(struct net_buf *buf);
+
+static inline void bt_buf_set_prio(struct net_buf *buf, bool prio)
+{
+	*((u8_t *)net_buf_user_data(buf) + 1) = prio;
+}
+
+static inline bool bt_buf_get_prio(struct net_buf *buf)
+{
+	return (bool)(*((u8_t *)net_buf_user_data(buf) + 1));
+}
```

Comment:
I'm not strictly opposed to having these, but now that you don't have a separate thread and queue anymore you could have made `bt_rpmsg_evt_recv()` return the priority instead of embedding it into the buffer user data, e.g. with an extra `bool *prio` parameter. Another problem with these is that they infringe on the bt_buf namespace, i.e. reading the calling code you might get an impression that they're public functions declared in `include/bluetooth/buf.h`

Reply:
done